src/app/infra/client-storage/local-storage.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import IClientStorage from './iclient-storage';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LocalStorageService implements IClientStorage {
+  set(key: string, data: unknown) {
+    if (key) {
+      try {
+        localStorage.setItem(key, JSON.stringify(data));
+      } catch (error) {
+        throw new Error(`Error saving to localStorage: ${error}`);
+      }
+      return;
+    }
+    throw new Error('Error saving to localStorage: key not provided');
+  }
+
+  get(key: string): unknown {
+    if (key) {
+      const value = localStorage.getItem(key);
+      if (value) {
+        return JSON.parse(value);
+      }
+      return null;
```

Comment:
why not throw an exception since it has no value in obj

Reply:
User shouldn't get an error trying to get some value that is not setted. I just kept the same return logic from `localStorage.getItem()` native method.